CONTRIBUTING.md

```diff
@@ -209,6 +209,72 @@ Before submitting your changes, run the test suite to make sure that nothing is
 ./gradlew check
 ```
 
+### Project layout
+
+This repository is split into many top level directories. The most important
+ones are:
+* docs: Documentation for the project.
+* distribution:  Builds our tar and zip archives and our rpm and deb packages.
+* libs: Libraries used to build other parts of the project. These are meant to
+        be internal rather than general purpose. We have no plans to
+        [semver](https://semver.org/) their APIs or accept feature requests
+        for them. We publish them to maven central because they are
+        dependencies of our plugin test framework, high level rest client, and
+        jdbc driver but.
+* modules:  Features that are shipped with Elasticsearch by default but are not
+        built in to the server. We typically separate features from the server
+        because they require permissions that we don't believe *all* of
+        Elasticsearch should have or because they depend on libraries that we
+        don't believe *all* of Elasticsearch should depend on. For example,
+        reindex requires the `connect` permission so it can perform
+        reindex-from-remote but we don't believe that the *all* of
+        Elasticsearch should have the "connect". For another example, Painless
+        is implemented using antlr4 and asm and we don't believe that *all*
+        of Elasticsearch should have access to them.
+* plugins:  Officially supported plugins to Elasticsearch. We decide that a
+        feature should be a plugin rather than built in to the Elasticsearch
+        server or shipped as a module because we feel that it is only important
+        to a subset of users, especially if it requires extra dependencies. The
+        canonical example of this is the ICU analysis plugin. It is important
+        for folks who want the fairly language neutral ICU analyzer but the
+        library to implement the analyzer is 11MB.
+* qa:  Honestly this is kind of in flux and we're not 100% sure where we'll end
```

Comment:
As a matter of taste, I'd also prefer it this has less of a "conversational" feel to it. Same vain as in the "we" case.

Reply:
I think being conversational is very important when we really don't know the right answer. I'm not sure I could be both formal and unsure. And I think it is important to communicate how we're unsure here.